authors/apps/articles/models.py

```diff
@@ -1,9 +1,67 @@
 from django.db import models
 from django.utils.text import slugify
+from django.db.models import Sum
+from django.utils.translation import ugettext_lazy as _
+from django.contrib.contenttypes.models import ContentType
+from django.contrib.contenttypes.fields import GenericRelation, GenericForeignKey
 
 from authors.apps.authentication.models import User
 
 
+class LikeDislikeManager(models.Manager):
+    """
+    Manager class for like and dislike
+    """
+    use_for_related_fields = True
+
+    def likes(self):
+        """
+        get all the likes for an object
+        """
+        return self.get_queryset().filter(vote__gt=0)
+
+    def dislikes(self):
+        """
+        get all the dislikes of an object
+        """
+        return self.get_queryset().filter(vote__lt=0)
+
+    def sum_rating(self):
+        """
+        obtain the aggregate likes/dislikes
+        """
+        return self.get_queryset().aggregate(Sum('vote')).get('vote__sum') or 0
+
+    def articles(self):
+        """
+        obtain the votes of a particular user
+        """
+        return self.get_queryset().filter(content_type__model='article').order_by('-articles__published_at')
+
+
+class LikeDislike(models.Model):
+    """
+    Users should be able to like and/or dislike an article or comment.
+    This class creates the fields for like/dislike
+    """
+    LIKE = 1
+    DISLIKE = -1
+
+    VOTES = (
+        (DISLIKE, 'Dislike'),
+        (LIKE, 'Like')
+    )
+
+    vote = models.SmallIntegerField(verbose_name=_("vote"), choices=VOTES)
```

Comment:
I like the use of choices here  and providing a data set that users will interact with alone